Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -538,6 +542,60 @@ void MemoryWidget::OnSetValue()
   Update();
 }
 
+void MemoryWidget::OnSetValueFromFile()
+{
+  if (!Core::IsRunning())
+    return;
+
+  bool good_address, good_offset;
+  // Returns 0 if conversion fails
+  u32 addr = m_search_address->text().toUInt(&good_address, 16);
+  addr += m_search_offset->text().toUInt(&good_offset, 16);
+
+  if (!good_address)
+  {
+    ModalMessageBox::critical(this, tr("Error"), tr("Bad address provided."));
+    return;
+  }
+
+  if (!good_offset)
+  {
+    ModalMessageBox::critical(this, tr("Error"), tr("Bad offset provided."));
+    return;
+  }
+
+  QString path = QFileDialog::getOpenFileName(this, tr("Select a file"), QDir::currentPath(),
+                                              QStringLiteral("All files (*.*)"));
+  if (path.isNull())
+  {
+    ModalMessageBox::critical(this, tr("Error"), tr("No file provided."));
+    return;
+  }
+
+  File::IOFile f(path.toStdString(), "rb");
+
+  if (f)
+  {
+    const u64 file_length = f.GetSize();
+    std::vector<u8> file_contents(file_length);
+    f.ReadBytes(file_contents.data(), file_length);
+    const QByteArray bytes(reinterpret_cast<const char*>(file_contents.data()), file_length);
+
+    AddressSpace::Accessors* accessors =
+        AddressSpace::GetAccessors(m_memory_view->GetAddressSpace());
+
+    for (const char c : bytes)
+      accessors->WriteU8(addr++, static_cast<u8>(c));
+  }
+  else
```

Comment:
This can be moved above in a `if (!f)` block instead. That way we can get rid of one layer of braces.
```c++
if (!f)
{
  ModalMessageBox::critical(nullptr, QObject::tr("Error"), QObject::tr("Unable to open file."));
  return;
}

const u64 file_length = f.GetSize();
std::vector<u8> file_contents(file_length);
// (rest of the code...)
```